cmd/crictl/container.go

```diff
@@ -367,18 +367,52 @@ var stopContainerCommand = &cli.Command{
 			Aliases: []string{"t"},
 			Usage:   "Seconds to wait to kill the container after a graceful stop is requested",
 		},
+		&cli.BoolFlag{
+			Name:    "all",
+			Aliases: []string{"a"},
+			Usage:   "Stop all running containers",
+		},
 	},
 	Action: func(c *cli.Context) error {
-		if c.NArg() == 0 {
-			return errors.New("ID cannot be empty")
-		}
 		runtimeClient, err := getRuntimeService(c, 0)
 		if err != nil {
 			return err
 		}
 
-		for i := range c.NArg() {
-			containerID := c.Args().Get(i)
+		if c.Bool("all") && c.NArg() > 0 {
+			return cli.Exit("cannot specify CONTAINER-ID(s) when using --all", 1)
```

Comment:
perhaps e2e test can be written for some of the error conditions like this in https://github.com/kubernetes-sigs/cri-tools/tree/master/test/e2e

Reply:
I felt that one would use -all to stop running containers than the one which are already stopped.